src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingCreateApprenticeFeedbackCommand.cs

```diff
@@ -0,0 +1,70 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ApprenticeFeedback.Application.Commands.CreateApprenticeFeedback;
+using SFA.DAS.ApprenticeFeedback.Domain.Entities;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using SFA.DAS.ApprenticeFeedback.Domain.Models;
+using SFA.DAS.Testing.AutoFixture;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.UnitTests.Commands
+{
+    public class WhenHandlingCreateApprenticeFeedbackCommand
+    {
+        [Test, MoqAutoData]
+        public async Task And_GetFeedbackTargetIsNull_Then_ThrowException
+            (CreateApprenticeFeedbackCommand command,
+            [Frozen] Mock<IApprenticeFeedbackRepository> mockApprenticeFeedbackRepository,
+            CreateApprenticeFeedbackHandler handler)
+        {
+            mockApprenticeFeedbackRepository.Setup(s => s.GetApprenticeFeedbackTargetById(command.ApprenticeId)).Throws(new Exception());
+
+            Func<Task> result = async () => await handler.Handle(command, CancellationToken.None);
+
+            await result.Should().ThrowAsync<Exception>();
+        }
+
+        [Test, MoqAutoData]
+        public async Task And_GetFeedbackTargetIsValid_And_GetAttributesIsNotValid_Then_ThrowException
+            (CreateApprenticeFeedbackCommand command,
+            [Frozen] Mock<IApprenticeFeedbackRepository> mockApprenticeFeedbackRepository,
+            CreateApprenticeFeedbackHandler handler,
+            Domain.Entities.ApprenticeFeedbackTarget apprenticeFeedbackTarget)
+        {
+            mockApprenticeFeedbackRepository.Setup(s => s.GetApprenticeFeedbackTargetById(command.ApprenticeId)).ReturnsAsync(apprenticeFeedbackTarget);
+            mockApprenticeFeedbackRepository.Setup(s => s.GetAttributes());
```

Comment:
This Setup method for GetAttributes doesn't actually do anything. We can either remove the line, and rely on the mock to never match what we supply in the command that's passed in .

Or we can be more specific, and set up some IEnumerable<Attribute> in the test constructor and setup the method to return those.